### PR TITLE
chore(ui): migrate ShadowRoot component to TypeScript

### PR DESCRIPTION
**Pull request by gjaskiewicz**
# Summary

Migrate ShadowRoot component to TypeScript

# Changes Made

- migrate ShadowRoot component to TypeScript

# Related Issues

- Issue 1: [[link to issue]](https://github.com/cloudoperators/juno/issues/250)

# Testing Instructions

1. `npm i`
2. `npm run storybook`

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

**Review by barsukov — CHANGES_REQUESTED** [review]
I have one question

File: packages/ui-components/src/components/ShadowRoot/ShadowRoot.component.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * The component automatically converts it to CSSStyleSheet,
+ * if supported by the browser. Otherwise there is a fallback to
+ * inline style element.
+ * @module ShadowRoot
+ */
+import ReactDOM from "react-dom"
+import React, { useRef, useState } from "react"
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// TODO: deactivating constructed stylesheets for now because of the decision of the authors to always include constructed stylesheets last
```
[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/ShadowRoot/ShadowRoot.component.tsx:19]: I think it is there: https://github.com/cloudoperators/juno/blob/c3ab4920878a363660b31e5018b04c07420f2f8e/packages/ui-components/src/components/ShadowRoot/ShadowRoot.component.js#L19

I didn't touched comments except of removing typings in comments as those are not in TS

[the remark this answers is not in the document]
Comment (barsukov) [reply packages/ui-components/src/components/ShadowRoot/ShadowRoot.component.tsx:19]: @edda I would like to ask you if we could just leave it like this. Please let us know what do you think.

[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/ShadowRoot/ShadowRoot.component.tsx:19]: @edda Would you be able to take a look?

**Review by barsukov — APPROVED** [review]

**Review by andypf — APPROVED** [review]
lgtm

**Pull request merged**